GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -425,5 +427,38 @@ private void OnNodeDoubleClick(object sender, TreeNodeMouseClickEventArgs e)
             // e.Node won't be the one you double clicked, but a child node instead
             Node.OnNode<Node>(treeMain.SelectedNode, node => node.OnDoubleClick());
         }
+
+        private void DisplayAheadBehindInformationForBranches()
+        {
+            if (_rootNodes.Count == 0)
+            {
+                return;
+            }
+
+            var aheadBehindData = new AheadBehindDataProvider(() => Module.GitExecutable).GetData();
```

Comment:
`AheadBehindDataProvider` is stateless, so we don't need to re-instantiate it every time.
Please check if it can be instantiated once and then reused, maybe at the time `Module` is set.